aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -286,8 +286,12 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
   }
 
   @Override
-  public List<TableIdentifier> listTables(Namespace namespace) {
-    namespaceExists(namespace);
+  public List<TableIdentifier> listTables(Namespace namespace) throws NoSuchNamespaceException {
```

Comment:
I wonder whether we need to handle empty namespaces here as well?

Reply:
Should we throw NoSuchNamespaceException incase of empty namespace because Glue doesn't support tables with-out a database or Should we return empty tables list considering empty namespace as a valid root namespace?

The behaviour in `listNamespaces(Namespace namespace)` returns all the glue databases in case of empty namespace, this may mean that empty namespace is a valid root namespace.

Note: empty namespaces implies namespace.isEmpty() is true in the context of nested namespaces.

Reply:
can you please add some tests for empty namespaces. Those should check the following:
* create / drop / namespace existence when given an empty namespace
* list namespaces via `listNamespaces()` and via `listNamespaces(Namespace.empty())`
* loading namespace metadata when given an empty namespace

I think it's currently not fully clear how empty namespaces are handled and it would be good to add these tests and have a consistent behavior when dealing with empty namespaces. This is mainly so that the check in this method does the right thing when an empty namespace is passed to `listTables()`